util/design/gen-otp-img.py

```diff
@@ -0,0 +1,183 @@
+# Default image file definition (can be overridden on the command line).
+IMAGE_DEFINITION_FILE = 'hw/ip/otp_ctrl/data/otp_ctrl_img_dev.hjson'
+# Default output path (can be overridden on the command line).
+MEMORY_HEXFILE = 'otp-img.mem'
+
+
+def _override_seed(args, name, config):
+    '''Override the seed key in config with value specified in args'''
+    arg_seed = getattr(args, name)
+    if arg_seed:
+        log.warning('Commandline override of {} with {}.'.format(
+            name, arg_seed))
+        config['seed'] = arg_seed
+    # Otherwise, we either take it from the .hjson if present, or
+    # randomly generate a new seed if not.
+    else:
+        random.seed()
+        new_seed = random.getrandbits(64)
+        if config.setdefault('seed', new_seed) == new_seed:
+            log.warning('No {} specified, setting to {}.'.format(
+                name, new_seed))
+
+
+def main():
+    log.basicConfig(level=log.INFO, format="%(levelname)s: %(message)s")
+
+    parser = argparse.ArgumentParser(
+        prog="gen-otp-img",
+        description=wrapped_docstring(),
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('--img-seed',
+                        type=int,
+                        metavar='<seed>',
+                        help='''
+                        Custom seed for RNG to compute randomized items in OTP image.
+
+                        Can be used to override the seed value specified in the image
+                        config Hjson.
+                        ''')
+    parser.add_argument('--lc-seed',
+                        type=int,
+                        metavar='<seed>',
+                        help='''
+                        Custom seed for RNG to compute randomized life cycle netlist constants.
+
+                        Note that this seed must coincide with the seed used for enerating
+                        the LC state encoding (gen-lc-state-enc.py).
+
+                        This value typically does not need to be specified as it is taken from
+                        the LC state encoding definition Hjson.
+                        ''')
+    parser.add_argument('--otp-seed',
+                        type=int,
+                        metavar='<seed>',
+                        help='''
+                        Custom seed for RNG to compute randomized otp netlist constants.
+
+                        Note that this seed must coincide with the seed used for enerating
+                        the OTP memory map (gen-otp-mmap.py).
+
+                        This value typically does not need to be specified as it is taken from
+                        the OTP memory map definition Hjson.
+                        ''')
+    parser.add_argument('-o',
+                        '--out',
+                        type=str,
+                        metavar='<path+filename>',
+                        default=MEMORY_HEXFILE,
```

Comment:
Maybe it would be worth doing some calculations based on `__file__` so that this can run from directories other than the top of the repository?